lib/api/rule.go

```diff
@@ -58,8 +62,12 @@ func init() {
 
 func validateRule(v *validator.Validate, structLevel *validator.StructLevel) {
 	rule := structLevel.CurrentStruct.Interface().(Rule)
-	if rule.ICMPCode != nil && rule.ICMPType == nil {
-		structLevel.ReportError(reflect.ValueOf(rule.ICMPCode), "ICMPCode", "icmpCode", "icmpCodeWithoutType")
+	if rule.ICMP != nil && rule.ICMP.Code != nil && rule.ICMP.Type == nil {
+		structLevel.ReportError(reflect.ValueOf(rule.ICMP.Code), "Code", "code", "icmpCodeWithoutType")
+	}
+
+	if rule.NotICMP != nil && rule.NotICMP.Code != nil && rule.NotICMP.Type == nil {
+		structLevel.ReportError(reflect.ValueOf(rule.NotICMP.Code), "Code", "code", "notIcmpCodeWithoutType")
```

Comment:
Looks to me like the fieldName / customName are used in the error reporting (but calicoctl isn't using them in the message it outputs).  I'd still update the tag as suggested above, and we can look at the other fields when we address #94.